src/enigma2/Timers.cpp

```diff
@@ -1195,7 +1209,14 @@ bool Timers::TimerUpdatesRegular()
 
   Logger::Log(LEVEL_DEBUG, "%s No of timers: removed [%d], untouched [%d], updated '%d', new '%d'", __FUNCTION__, iRemoved, iUnchanged, iUpdated, iNew);
 
-  std::vector<EpgEntry> timerBaseEntries(m_timers.begin(), m_timers.end());
+  std::vector<EpgEntry> timerBaseEntries;
+  for (auto& timer : m_timers)
+  {
+    EpgEntry entry = dynamic_cast<EpgEntry&>(timer);
```

Comment:
The return value of dynamic_cast should always be checked as it might fail. Static code analysis tools (like Coverity) complain about this, for example.

Reply:
There is no real reason to use dynamic cast here anyway. static should work just fine.

Reply:
Do we actually need a cast here? Timer is an EpgEntry , right?

Reply:
Ya, you're right ;)